packages/graphback-keycloak-authz/README.md

```diff
@@ -0,0 +1,285 @@
+
+## Example with Graphback Runtime
+
+During server initialization, use the `createKeycloakRuntimeContext` function to initialize the KeycloakCrudService instances for each model.
+
+The following example shows just the necessary parts to set up the runtime services in Graphback.
+
+```ts
+import { ApolloServer } from 'apollo-server-express'
+import { createKeycloakRuntimeContext } from '@graphback/keycloak-authz'
+import { GraphbackRuntime } from 'graphback'
+import { KnexDbDataProvider } from '@graphback/runtime-knex'
+import { PubSub } from 'graphql-subscriptions'
+import * as Knex from 'knex'
+import { printSchema } from 'graphql'
+
+// the application model
+const model = `
+""" 
+@model
+"""
+type Task {
+  id: ID!
+  title: String!
+  description: String!
+}`
+
+// the auth rules for the application
+const authConfig = {
+  Task: {
+    create: {},
+    read: {},
+    update: { roles: ["admin"] },
+    delete: { roles: ["admin"] }
+  }
+}
+
+// set up the Knex database client
+const db = Knex({...})
+
+// standard Graphback runtime setup
+const pubSub = new PubSub()
+const runtimeEngine = new GraphbackRuntime(model, graphbackConfig)
+const models = runtimeEngine.getDataSourceModels()
+
+// create runtime services integrated with Keycloak
+// Note that the knex database client and the KnexDBDataProvider class are passed
+// Other data providers can be passed too
+// Under the hood this creates KeycloakCrudService instances
+const services = createKeycloakRuntimeContext({ 
+  models,
+  schema: model,
+  db,
+  pubSub,
+  authConfig,
+  dataProvider: KnexDBDataProvider
+})
+
+const { schema, resolvers } = runtimeEngine.buildRuntime(services);
+
+const server = new ApolloServer({
+  typeDefs: printSchema(schema),
+  resolvers,
+  context: (req) => {
+    return {
+      kauth: new KeycloakContext({ req })
+    }
+  }
+})
+```
+
+The above example shows runtime set up using the KnexDbDataProvider, but other data providers such as the `MongoDBDataProvider` can also be passed.
+
+## Extending the KeycloakCrudService
+
+Under the hood, the `createKeycloakRuntimeContext` initializes `KeycloakCrudService` instances. The `KeycloakCrudService` extends the standard `CrudService` in Graphback and adds additional checks for various auth rules.
+
+It is possible to create a custom CrudService that extends the KeycloakCrudService and pass it to `createKeycloakRuntimeContext`.
+
+```ts
+import { KeycloakCrudService } from '@graphback/keycloak-authz'
+
+class CustomCrudService extends KeycloakCrudService {
+  // custom implementation
+}
+
+const services = createKeycloakRuntimeContext({ 
+  models,
+  schema: model,
+  db,
+  pubSub,
+  authConfig,
+  dataProvider: KnexDBDataProvider,
+  crudService: CustomCrudService
+})
+```
+
+## Full Example showing Keycloak integration in apollo-server-express
+
+The following example is a full that shows how to set up a Graphback runtime application that uses Keycloak authentication and authorization.
+
+```ts
```

Comment:
I would put that into file for better maintenance :)
Example will be anyway in:

- DataSync repo
- DataSync starter
- OVP 

so this could be great place for reference. We will want to add that to OVP after release and push some short blog post about auth approach. 

This docs works perfectly. We will need 2-5 sentences introduction and we can push that as new blog